JamesIves/github-pages-deploy-action
# Native SSH Key Support

**Description**
> Provide a description of what your changes do.

Implements SSH key support natively to the project.

**Testing Instructions**
> Give us step by step instructions on how to test your changes.

* Generate an SSH key: `ssh-keygen -t rsa -m pem -b 4096 -C "youremailhere@example.com" -N ""`
* Add the public key as a deploy token in the GitHub repo, give it write access.
* Add the private key as a secret using the `ssh-key` input.
* Attempt to deploy, the token should be added and the project will deploy.

**Additional Notes**
> Anything else that will help us test the pull request.

There's a few things that need to be wrapped up, but I wanted to get this to PR sooner as I've been holding up v4 with this.

* [x] Fix TODO statements in constants.ts.
* [x] Add sshKey to action.yml
* [x] Fix README.
* [x] Add/fix unit tests for `ssh.ts`.
